src/api/init.ts

```diff
@@ -89,9 +95,58 @@ export async function startApiServer(datastore: DataStore): Promise<ApiServer> {
     })
   );
 
-  const server = await new Promise<Server>((resolve, reject) => {
+  const dbTxUpdate = async (txId: string): Promise<void> => {
+    if (tx_subscribers.has(txId)) {
+      try {
+        const txQuery = await getTxFromDataStore(txId, datastore);
+        if (!txQuery.found) {
+          throw new Error('error in tx stream, tx not found');
+        }
+        tx_subscribers
+          .get(txId)
+          ?.forEach(subscriber =>
+            sendWsTxUpdate(subscriber, txQuery.result.tx_id, txQuery.result.tx_status)
+          );
+      } catch (error) {
+        logError('error streaming tx updates', error);
+      }
+    }
+  };
+
+  // EventEmitters don't like being passed Promise functions so wrap the async handler
+  const onTxUpdate = (txId: string): void => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    dbTxUpdate(txId);
+  };
+
+  datastore.addListener('txUpdate', onTxUpdate);
+
+  let server = createServer(app);
+  const wss = new WebSocket.Server({ server, path: '/sidecar/v1' });
+  wss.on('connection', function (ws) {
+    ws.on('message', txid => {
+      const id = txid.toString();
+      const connections = tx_subscribers.get(id);
```

Comment:
Nit: snake_case is only used to match JSON schemas and sql names, the rest should all be camelCase